pkg/client/cache/listers.go

```diff
@@ -58,17 +59,92 @@ func (s *StoreToWorkflowLister) GetJobWorkflows(job *k8sBatch.Job) (workflows []
 	}
```

Comment:
It could be more than one workflow, alltough it's not expected bahaviour.
workflow/manager.go has a function getJobWorkflow (singular), which calls this function and checks the amount of workflows returned.